src/hooks/persisted/useNovel.ts

```diff
@@ -235,6 +269,19 @@ export const useNovel = (novelPath: string, pluginId: string) => {
   };
 
   const markChaptersRead = (_chapters: ChapterInfo[]) => {
+    if (novel) {
+      // Track progress if the plugin supports it
+      const plugin = getPlugin(novel.pluginId);
+      if (plugin?.trackProgress) {
+        // Sort the selected chapters based on the position
+        const sortedChapters = [..._chapters].sort((a, b) => {
+          return b.position! - a.position!;
+        });
+        const trackedChapter = sortedChapters[0];
```

Comment:
It is easier to see what it does that way.
```suggestion
        const trackedChapter = _chapters.reduce((prev: ChapterInfo, curr: ChapterInfo)=>{
                 if (curr.position > prev.position) return curr
                 return prev
                 }
          )
```